interface-impl-amf/src/main/java/org/mule/amf/impl/AMFParser.java

```diff
@@ -126,58 +131,29 @@ public WebApi getWebApi() {
 
   @Override
   public ApiValidationReport validate() {
-    ValidationReport validationReport = generateValidationReport();
+    ValidationReport validationReport = getParsingReport(parser.getParser(), parser.getProfileName());
     List<ApiValidationResult> results = new ArrayList<>(0);
     if (!validationReport.conforms()) {
       results = validationReport.results().stream().map(ApiValidationResultImpl::new).collect(toList());
     }
     return new DefaultApiValidationReport(results);
   }
 
-  private ValidationReport generateValidationReport() {
-    final ValidationReport validationReport;
-    try {
-      validationReport = parser.reportValidation(apiVendorToProfileName(apiRef.getVendor())).get();
-    } catch (InterruptedException | ExecutionException e) {
-      throw new RuntimeException("Unexpected error parsing API: " + e.getMessage(), e);
-    }
-    return validationReport;
-  }
-
-  private ProfileName apiVendorToProfileName(ApiVendor apiVendor) {
-    switch (apiVendor) {
-      case OAS:
-        return OAS();
-      case OAS_20:
-        return OAS20();
-      case RAML:
-        return RAML();
-      case RAML_08:
-        return RAML08();
-      case RAML_10:
-        return RAML10();
-      default:
-        return AMF();
-    }
-  }
-
   @Override
   public ApiSpecification parse() {
     return new AMFImpl(webApi, references, apiRef.getVendor(), new LazyValue<>(
         this::getConsoleModel), apiRef);
   }
 
   private Document getConsoleModel() {
-
     if (consoleModel == null) {
-      Document document = buildDocument(false);
-      consoleModel = (Document) new Raml10Resolver().resolve(document, "editing");
```

Comment:
can we remove this ?

Reply:
I don't get you question. The code that is being removed here was moved to ```DocumentParser.parseFile```. For the contrary, If you are wondering if we can get rid of the whole method, I don't think so because this is being called from the LazyValue that builds the console model.